### PR TITLE
Add attestation clients

### LINKED ISSUE #3444: Add authentication sdk to smoke tests

### PR DESCRIPTION
**Pull request by gearama**
closes #3444 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 1]
```diff
@@ -25,51 +26,62 @@ using namespace Azure::Storage::Blobs;
...
+    AttestationClient attestationClient(smokeUrl);
+    AttestationAdministrationClient attestationAdminClient(smokeUrl, credential);
+
+    std::cout << "Successfully Created the Clients"
+              << "\n";
   }
   catch (std::exception const& exception)
   {
-    std::cout << "Exception: " << exception.what();
+    std::cout << "Exception: " << exception.what() << "\n";
```
Comment (LarryOsterman) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:84]:
```suggestion
    std::cout << "Exception: " << exception.what() << std::endl;
```

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 2]
```diff
@@ -25,51 +26,62 @@ using namespace Azure::Storage::Blobs;
...
+    ShareServiceClient shareServiceClient(smokeUrl);
 
-    std::cout << "Successfully Created the Clients";
+    std::cout << "Creating Storage Attestation Clients"
+              << "\n";
+    AttestationClient attestationClient(smokeUrl);
+    AttestationAdministrationClient attestationAdminClient(smokeUrl, credential);
+
+    std::cout << "Successfully Created the Clients"
+              << "\n";
```
Comment (LarryOsterman) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:80]:
```suggestion
              << std::endl;
```

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 3]
```diff
@@ -25,51 +26,62 @@ using namespace Azure::Storage::Blobs;
...
+    BlobClient blobClient(smokeUrl);
+    QueueClient queueClient(smokeUrl);
 
-    std::cout << "Creating Storage Datalake Clients";
-    DataLakeDirectoryClient directoryClient(storageUrl);
-    DataLakeFileClient fileClient(storageUrl);
-    DataLakeFileSystemClient fileSystemClient(storageUrl);
-    DataLakePathClient pathClient(storageUrl);
+    std::cout << "Creating Storage Datalake Clients"
+              << "\n";
```
Comment (LarryOsterman) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:58]:
```suggestion
              << std::endl;
```

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 4]
```diff
@@ -25,51 +26,62 @@ using namespace Azure::Storage::Blobs;
...
+  const std::string smokeUrl = "https://blob.com";
   auto credential
       = std::make_shared<Azure::Identity::ClientSecretCredential>(tenantId, clientId, clientSecret);
 
   // instantiate the clients
   try
   {
-    std::cout << "Creating Keyvault Clients";
+    std::cout << "Creating Keyvault Clients"
+              << "\n";
```
Comment (LarryOsterman) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:45]:
```suggestion
              << std::endl;
```

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 5]
```diff
@@ -25,51 +26,62 @@ using namespace Azure::Storage::Blobs;
...
     DataLakeLeaseClient leaseClient(pathClient, leaseID);
-    DataLakeServiceClient serviceClient(storageUrl);
+    DataLakeServiceClient serviceClient(smokeUrl);
 
-    std::cout << "Creating Storage Share Clients";
-    ShareClient shareClient(storageUrl);
-    ShareDirectoryClient shareDirectoryClient(storageUrl);
-    ShareFileClient shareFileClient(storageUrl);
+    std::cout << "Creating Storage Share Clients"
+              << "\n";
```
Comment (LarryOsterman) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:67]:
```suggestion
              << std::endl;
```

**Review by LarryOsterman — DISMISSED** [review]

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 6]
```diff
@@ -25,51 +26,57 @@ using namespace Azure::Storage::Blobs;
...
+    std::cout << "Creating Attestation Clients" << std::endl;
+    AttestationClient attestationClient(smokeUrl);
+    AttestationAdministrationClient attestationAdminClient(smokeUrl, credential);
+
+    std::cout << "Successfully Created the Clients" << std::endl;
   }
   catch (std::exception const& exception)
   {
-    std::cout << "Exception: " << exception.what();
+    std::cout << "Exception: " << exception.what() << std::endl;
```
Comment (vhvb1989) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:79]:
```suggestion
    std::cout << "Exception: " << exception.what() << std::endl;
    return 1;
```

File: samples/integration/vcpkg-all-smoke/src/main.cpp [hunk 7]
```diff
@@ -25,51 +26,57 @@ using namespace Azure::Storage::Blobs;
 using namespace Azure::Storage::Queues;
 using namespace Azure::Storage::Files::DataLake;
 using namespace Azure::Storage::Files::Shares;
+using namespace Azure::Security::Attestation;
 
 int main()
```
Comment (vhvb1989) [inline samples/integration/vcpkg-all-smoke/src/main.cpp:31]: I won't complain about program, I think it brings value.
However, I do think the value it brings is super minimal and basically reduced to `check we can compile clients into one binary`
Remember the linker will take only what's used by the application, so we are only taking the client definitions (hpp files and the constructor implementation).

Hopefully we could call operations from the clients to see them all calling I/O one after eachother and also at the same time.  That would be fun!

**Review by vhvb1989 — APPROVED** [review]

File: samples/integration/vcpkg-all-smoke/src/main.cpp (hunk 7 above)
Comment (gearama) [reply samples/integration/vcpkg-all-smoke/src/main.cpp:31]: initially i was doing that , but after some discussion with some people involved in this, the decission was made to not call any APIs.

**Review by LarryOsterman — APPROVED** [review]

**Pull request merged**